.pre-commit-config.yaml

```diff
@@ -1,41 +1,50 @@
+ci:
+  autofix_prs: false
+
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-merge-conflict
+      - id: check-case-conflict
+      - id: check-toml
+      - id: check-json
+      - id: check-ast
+      - id: debug-statements
+
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+      # - id: python-check-blanket-noqa # XXX: disallow blanket # noqa
+      - id: python-check-blanket-type-ignore
+      - id: python-check-mock-methods
+      # - id: python-use-type-annotations # XXX: disallow type comments
+
+  - repo: https://github.com/pycqa/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+        types: [python] # FIXME: pyi support that doesn't conflcit with black
+
   - repo: https://github.com/psf/black
-    rev: 20.8b1
+    rev: 21.10b0
     hooks:
       - id: black
 
   - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.8.4
+    rev: 4.0.1
     hooks:
       - id: flake8
-        additional_dependencies: [flake8-bugbear]
+        additional_dependencies:
+          # - flake8-docstring # XXX: lint docstrings
+          - flake8-bugbear
+          # - pep8-naming # XXX: lint names
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.910
+    rev: v0.910-1
     hooks:
       - id: mypy
         pass_filenames: false
         additional_dependencies:
           - types-requests
-        args:
-          - poetry
-
-  - repo: https://github.com/timothycrosley/isort
-    rev: 5.7.0
-    hooks:
-      - id: isort
-        additional_dependencies: [toml]
-        exclude: ^.*/?setup\.py$
 
-  - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.4.0
-    hooks:
-      - id: trailing-whitespace
```

Comment:
These are gone because `black` provides them for Python files. If they're desired for other file types they can easily be added back.